easybuild/framework/easyblock.py

```diff
@@ -1864,6 +1866,8 @@ def skip_extensions_parallel(self, exts_filter):
 
         self.ext_instances = retained_ext_instances
 
+        thread_pool.shutdown()
```

Comment:
Maybe we can avoid an explicit shutdown by launching the executor through a `with` and use `as_completed` instead of `wait` . We can move the call to `ThreadPoolExecutor` lower to keep things compact as well:

Following the spirit of my previous comment with the 2 independent loops:
```
cmd_queue = [resolve_exts_filter_template(exts_filter, ext) for ext in exts_queue
with ThreadPoolExecutor(max_workers=self.cfg['parallel']) as thread_pool:
    async_shell_cmd_tasks = [thread_pool.submit(run_shell_cmd, cmd, ...) for cmd, stdin in cmd_queue]
    for done_task in concurrent.futures.as_completed(async_shell_cmd_tasks):
        res = done_task.result()
        idx = res.task_id
        [...]
        self.update_exts_progress_bar(exts_pbar_label)

# compose new list of extensions, skip over the ones that are already installed;
[...]

Reply:
fixed in ca3828b